### PR TITLE
Conditional process on changed methods

### PR DESCRIPTION
**Pull request by safoster88**
Added a conditional check to ProcessOnChangedMethods call, so that it is not checked in the circumstance that InjectOnPropertyNameChanged = false.

This solves the issue of compilation failing when the requirements for OnPropertyChanged injection are not met when InjectOnPropertyNameChanged is false.

Know I'm not a patron yet, but I thought this might be helpful (It might also be completely wrong)

### CODE REVIEW

**Comment by SimonCropp** [comment]
Can you add a test

**Comment by safoster88** [comment]
Added a couple of tests to verify the change.

File: Tests/OnPropertyNameChangedConfigTests.cs [hunk 1]
```diff
@@ -31,6 +32,31 @@ public void Default()
         Assert.True(moduleWeaver.InjectOnPropertyNameChanged);
     }
 
+    [Fact]
+    public void ModuleWeaver_WhenInjectOnPropertyNameChangedIsTrue_ThrowsWeavingException()
```
Comment (SimonCropp) [inline Tests/OnPropertyNameChangedConfigTests.cs]: this one seems to pass even without your change

**Comment by SimonCropp** [comment]
OnPropertyNameChangedConfigTests.cs is about verifying the config is read correctly, which i dont think is what you are trying to verify

File: Tests/OnPropertyNameChangedConfigTests.cs [hunk 2]
```diff
@@ -31,6 +32,31 @@ public void Default()
...
+
+        Assert.Equal("The type ClassWithNonVoidOnPropertyChanged has a On_PropertyName_Changed method (OnProperty1Changed) that has a non void return value. Ensure the return type void.", weavingException.Message);
+    }
+
+    [Fact]
+    public void ModuleWeaver_WhenInjectOnPropertyNameChangedIsFalse_DoesNotThrowWeavingException()
+    {
+        var xElement = XElement.Parse("<PropertyChanged InjectOnPropertyNameChanged='false'/>");
+        var moduleWeaver = new ModuleWeaver { Config = xElement };
+        moduleWeaver.ResolveOnPropertyNameChangedConfig();
```
Comment (SimonCropp) [inline Tests/OnPropertyNameChangedConfigTests.cs]: these 3 lines can be replaced with
```
var moduleWeaver = new ModuleWeaver { InjectOnPropertyNameChanged = false};
```

File: Tests/OnPropertyNameChangedConfigTests.cs (hunk 1 above)
Comment (safoster88) [reply Tests/OnPropertyNameChangedConfigTests.cs]: Yeah that's right as it verifies that the error is displayed when the Inject property is true.

Before the error would be displayed in both true & false.

I can remove this one if you like?

Comment (SimonCropp) [reply Tests/OnPropertyNameChangedConfigTests.cs]: Fair enough. Let’s keep it

File: Tests/OnPropertyNameChangedConfigTests.cs (hunk 2 above)
Comment (safoster88) [reply Tests/OnPropertyNameChangedConfigTests.cs]: Done that,

Cheers!

**Comment by SimonCropp** [comment]
This is now deployed. NuGet may take some time to make it available for download.

**Pull request merged**